modules/metastorage/src/main/java/org/apache/ignite/internal/metastorage/server/WatchProcessor.java

```diff
@@ -266,7 +266,14 @@ private CompletableFuture<Void> notifyWatchesInternal(long newRevision, List<Ent
             newNotificationFuture.whenComplete((unused, e) -> maybeLogLongProcessing(filteredUpdatedEntries, startTimeNanos));
 
             return newNotificationFuture;
-        });
+        }, updatedEntriesKeysInfo(updatedEntries));
+    }
+
+    private static String updatedEntriesKeysInfo(List<Entry> updatedEntries) {
+        String joinedKeys = updatedEntries.stream()
+                .map(entry -> new String(entry.key(), UTF_8))
+                .collect(joining(","));
+        return "Keys of updated entries: " + joinedKeys;
```

Comment:
`"Keys of updated entries: "` can be passed directly into `joining` as a prefix value